libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -81,25 +77,28 @@ internal class MapboxTripSession(
         legIndex: Int,
         @RoutesExtra.RoutesUpdateReason reason: String
     ) {
-
-        isOffRoute = false
-        invalidateLatestInstructions()
-        roadObjects = emptyList()
-        routeProgress = null
-
         updateRouteJob = when (reason) {
             RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP,
             RoutesExtra.ROUTES_UPDATE_REASON_NEW,
-            RoutesExtra.ROUTES_UPDATE_REASON_REROUTE,
-            RoutesExtra.ROUTES_UPDATE_REASON_ALTERNATIVE -> {
+            RoutesExtra.ROUTES_UPDATE_REASON_REROUTE -> {
+                isOffRoute = false
+                invalidateLatestInstructions()
+                roadObjects = emptyList()
+                routeProgress = null
                 updateLegIndexJob?.cancel()
                 updateRouteProgressJob?.cancel()
 
                 threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
-                    nativeRouteProcessingListeners.forEach { it.onNativeRouteProcessingStarted() }
-                    navigator.setRoute(routes, legIndex)?.let {
+                    navigator.setPrimaryRoute(routes.firstOrNull(), legIndex)?.let {
                         roadObjects = getRouteInitInfo(it)?.roadObjects ?: emptyList()
                     }
+                    navigator.setAlternativeRoutes(routes.drop(1))
```

Comment:
🚨  Does it mean that `TripSession` doesn't process status update until navigator set alternatives?

Reply:
That's correct. On par with what we had before but there is some room for improvement.

Reply:
@DenisPryt , as far as I remember you told me that native navigator starts publishing updates as soon as we set primary route. Can you please confirm?

Reply:
Right. setPrimaryRoute starts status reporting.

Reply:
It seems that we can miss first status updates while waiting for `setAlternativeRoutes` to complete

Reply:
Fixed!